pkg/controller/daemon/daemon_controller.go

```diff
@@ -1255,47 +1255,58 @@ func (dsc *DaemonSetsController) syncDaemonSet(ctx context.Context, key string)
 }
 
 // NodeShouldRunDaemonPod checks a set of preconditions against a (node,daemonset) and returns a
-// summary. Returned booleans are:
-// * shouldRun:
-//     Returns true when a daemonset should run on the node if a daemonset pod is not already
-//     running on that node.
-// * shouldContinueRunning:
-//     Returns true when a daemonset should continue running on a node if a daemonset pod is already
-//     running on that node.
-func NodeShouldRunDaemonPod(node *v1.Node, ds *apps.DaemonSet) (bool, bool) {
+// summary. Returned errors are:
+// * shouldNotRun:
+//     Returns nil when a daemonset should run on the node if a daemonset pod is not already
+//     running on that node.  Otherwise returns an error explaining why not.
+// * shouldNotContinueRunning:
+//     Returns nil when a daemonset should continue running on a node if a daemonset pod is already
+//     running on that node.  Otherwise returns an error explaining why not.
+func NodeShouldRunDaemonPod(node *v1.Node, ds *apps.DaemonSet) (error, error) {
```

Comment:
nit: Returning "why not" sounds good but it doesn't feel idiomatic to have multiple errors returned.

SIG apps folks can probably better tell how useful the change is.

Reply:
I'm just thinking, should they still be `shouldRun` and `shouldContinueRunning`? The expected value `nil` means "should" and `error` means "should not" with reasons. So does the `predicates` below.